pkg/connectors/datacatalog/clients/datacatalog_openapi.go

```diff
@@ -48,33 +56,45 @@ func NewOpenAPIDataCatalog(name, connectionURL string) DataCatalog {
 }
 
 func (m *openAPIDataCatalog) GetAssetInfo(in *datacatalog.GetAssetRequest, creds string) (*datacatalog.GetAssetResponse, error) {
-	resp, _, err := m.client.DefaultApi.GetAssetInfo(context.Background()).XRequestDatacatalogCred(creds).GetAssetRequest(*in).Execute()
+	resp, httpResponse, err :=
+		m.client.DefaultApi.GetAssetInfo(context.Background()).XRequestDatacatalogCred(creds).GetAssetRequest(*in).Execute()
+	defer httpResponse.Body.Close()
+	if httpResponse.StatusCode == http.StatusForbidden {
+		return nil, errors.New(AccessForbidden)
+	}
+	if httpResponse.StatusCode == http.StatusNotFound {
+		return nil, errors.New(InvalidAssetID)
+	}
 	if err != nil {
 		return nil, errors.Wrap(err, fmt.Sprintf("get asset info from %s failed", m.name))
 	}
 	return resp, nil
 }
 
 func (m *openAPIDataCatalog) CreateAsset(in *datacatalog.CreateAssetRequest, creds string) (*datacatalog.CreateAssetResponse, error) {
-	resp, _, err := m.client.DefaultApi.CreateAsset(context.Background()).
+	resp, httpResponse, err := m.client.DefaultApi.CreateAsset(context.Background()).
 		XRequestDatacatalogWriteCred(creds).CreateAssetRequest(*in).Execute()
+	defer httpResponse.Body.Close()
 	if err != nil {
 		return nil, errors.Wrap(err, fmt.Sprintf("create asset info from %s failed", m.name))
 	}
 	return resp, nil
 }
 
 func (m *openAPIDataCatalog) DeleteAsset(in *datacatalog.DeleteAssetRequest, creds string) (*datacatalog.DeleteAssetResponse, error) {
-	resp, _, err := m.client.DefaultApi.DeleteAsset(context.Background()).XRequestDatacatalogCred(creds).DeleteAssetRequest(*in).Execute()
+	resp, httpResponse, err :=
+		m.client.DefaultApi.DeleteAsset(context.Background()).XRequestDatacatalogCred(creds).DeleteAssetRequest(*in).Execute()
```

Comment:
httpResponse should be checked right?

Reply:
Added a check for asset not found. All responses that are not successful return an error which is checked.